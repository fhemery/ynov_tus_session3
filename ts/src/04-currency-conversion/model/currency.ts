export enum Currency {
  Euro = "€",
  Dollar = "$",
  Pound = "£",
  Yen = "¥"
}