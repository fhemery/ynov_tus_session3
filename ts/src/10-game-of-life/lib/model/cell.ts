export class Cell {
  constructor(readonly row: number, readonly col: number) {

  }

}
