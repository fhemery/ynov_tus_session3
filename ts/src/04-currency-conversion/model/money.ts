import { Currency } from "./currency";

export class Money {
  constructor(readonly amount: number, readonly currency: Currency) {
  }
}