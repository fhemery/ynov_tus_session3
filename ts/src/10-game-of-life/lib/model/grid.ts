import {Cell} from "./cell";

export class Grid {
  private _cells: boolean[][];
  get aliveCells(): Cell[] {
    const result = [];
    for(let row = 0; row < this.rows; ++row) {
      for(let col = 0; col < this.cols; ++col) {
        if(this._cells[row][col]) {
          result.push(new Cell(row+1, col+1));
        }
      }
    }
    return result;
  }

  constructor(readonly rows: number, readonly cols: number) {
    this._cells = Array.from({length: rows}, () => Array.from({length: cols}, () => false));
  }

  withAliveCell(row: number, col: number): Grid {
    this._cells[row-1][col-1] = true;
    return this;
  }
}
