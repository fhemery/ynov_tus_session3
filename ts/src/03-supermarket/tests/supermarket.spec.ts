import { Teller } from "../src/model/Teller";
import { FakeCatalog } from "./FakeCatalog";
import { ShoppingCart } from "../src/model/ShoppingCart";
import { Product } from "../src/model/Product";
import { ProductUnit } from "../src/model/ProductUnit";

describe("Supermarke - Teller", function() {

  let teller: Teller;

  it("should return nothing on empty basket", function() {
    // ARRANGE
    teller = new Teller(new FakeCatalog());
    const shoppingCart = new ShoppingCart();
    
    // ACT
    const receipt = teller.checksOutArticlesFrom(shoppingCart);

    // ASSERT
    expect(receipt.getItems()).toHaveLength(0);
    expect(receipt.getDiscounts()).toHaveLength(0);

    expect(receipt).toMatchSnapshot();
  });

  it('should return one item and no discount', () => {
    // ARRANGE
    const catalog = new FakeCatalog();

    const apples = new Product("Apples", ProductUnit.Kilo)
    catalog.addProduct(apples, 5);

    const shoppingCart = new ShoppingCart();
    shoppingCart.addItemQuantity(apples, 2);

    teller = new Teller(catalog);
    // ACT
    const receipt = teller.checksOutArticlesFrom(shoppingCart);

    // ASSERT
    /*expect(receipt.getItems()).toHaveLength(1);
    expect(receipt.getItems()[0].product.name).toBe("Apples");
    expect(receipt.getItems()[0].totalPrice).toBe(10);
    expect(receipt.getTotalPrice()).toBe(10);*/

    expect(receipt).toMatchSnapshot();
  })
});