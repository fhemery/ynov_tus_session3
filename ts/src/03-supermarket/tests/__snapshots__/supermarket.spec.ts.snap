// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Supermarke - Teller should return nothing on empty basket 1`] = `
Receipt {
  "discounts": Array [],
  "items": Array [],
}
`;

exports[`Supermarke - Teller should return one item and no discount 1`] = `
Receipt {
  "discounts": Array [],
  "items": Array [
    ReceiptItem {
      "price": 5,
      "product": Product {
        "name": "Apples",
        "unit": 0,
      },
      "quantity": 2,
      "totalPrice": 10,
    },
  ],
}
`;
